src/cli_util.rs

```diff
@@ -962,12 +964,41 @@ impl WorkspaceCommandHelper {
         }
     }
 
+    /// Resolve a revset any number of revisions (including 0).
     pub fn resolve_revset(&self, revision_str: &str) -> Result<Vec<Commit>, CommandError> {
         let revset_expression = self.parse_revset(revision_str)?;
         let revset = self.evaluate_revset(revset_expression)?;
         Ok(revset.iter().commits(self.repo().store()).try_collect()?)
     }
 
+    /// Resolve a revset any number of revisions (including 0), but require the
+    /// user to indicate if they allow multiple revisions by wrapping the
+    /// expression in `multiple()`.
+    pub fn resolve_revset_default_single(
+        &self,
+        revision_str: &str,
+    ) -> Result<Vec<Commit>, CommandError> {
+        let revset_expression = self.parse_revset(revision_str)?;
+        if matches!(revset_expression.as_ref(), RevsetExpression::Multiple(_)) {
+            self.resolve_revset(revision_str)
+        } else {
```

Comment:
I'm not sure if `multiple(none())` should be allowed. If `multiple(x)` is equivalent to `sized(x, 1..)`, empty set should be rejected here, not by the caller. If `multiple(x)` is just a maker to lift implicit `sized(x, 1)`, no check should be made here.

fwiw, I have draft commit that implements `sized()`.

Reply:
The way it currently works is that it's like `sized(x, 0..)`, i.e. allowing any size, and then the caller checks that the set is not empty if it requires that. Do you think it's better to make `multiple()` require at least one?

Even when we have `sized()`, I suspect we won't actually have the CLI wrap the parsed expression in `sized(x, 1..)` because it would probably make error messages less clear, or what do you think?

Reply:
> Do you think it's better to make `multiple()` require at least one?

My feeling is that `multiple` means `> 1` or `>= 1`, but implementation wise, `>= 0` is the easiest. I think the desired behavior is more like `unsized()` or `unbound()`, but I don't know if such term is better.

> Even when we have `sized()`, I suspect we won't actually have the CLI wrap the parsed expression in `sized(x, 1..)` because it would probably make error messages less clear

Sure. In order to replace the CLI usage, we'll need a mechanism to attach the source expression to runtime error.

Reply:
> My feeling is that `multiple` means `> 1` or `>= 1`, but implementation wise, `>= 0` is the easiest. I think the desired behavior is more like `unsized()` or ~`unbound()`~ (EDIT: `unbounded()`?), but I don't know if such term is better.

Yes, I agree with that, but I also couldn't think of a better name. `unbounded()` sounds pretty good.